File: android/app/src/main/java/com/vxin/app/data/model/Group.kt
package com.vxin.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class GroupMember(
    val id: String,
    val username: String = "",
    val avatar: String = "",
    val role: String = "member",      // owner | admin | member
    val nickname: String? = null,
) {
    val displayName: String get() = nickname?.takeIf { it.isNotBlank() } ?: username
}

/** GET conversation/{id}/info —— conv 字段 + members + myRole（Json 忽略未知字段） */
@Serializable
data class GroupInfo(
    val id: String,
    val name: String = "",
    val avatar: String = "",
    val announcement: String = "",
    val owner_id: String = "",
    val myRole: String = "member",
    val mute_all: Int = 0,
    val no_private_chat: Int = 0,
    val no_add_friend: Int = 0,
    val members: List<GroupMember> = emptyList(),
) {
    val canManage: Boolean get() = myRole == "owner" || myRole == "admin"
    val isOwner: Boolean get() = myRole == "owner"
    fun myNickname(myId: String): String = members.firstOrNull { it.id == myId }?.nickname.orEmpty()
}

/** 群管理设置（PUT .../manage，群主/管理员） */
@Serializable
data class ManageBody(
    val mute_all: Boolean? = null,
    val no_private_chat: Boolean? = null,
    val no_add_friend: Boolean? = null,
)

/** 设置成员角色（PUT .../members/:uid/role，仅群主） */
@Serializable
data class SetRoleBody(val role: String)   // admin | member

@Serializable
data class RenameGroupBody(val name: String)

/** 更新群信息（群名 / 群公告，仅群主、管理员） */
@Serializable
data class UpdateGroupBody(val name: String? = null, val announcement: String? = null)

/** 设置我的群昵称 */
@Serializable
data class NicknameBody(val nickname: String)

@Serializable
data class InviteBody(val userIds: List<String>)

File: ios/Vxin/Data/Models/Group.swift
import Foundation

struct GroupMember: Decodable, Identifiable, Hashable {
    let id: String
    var username: String = ""
    var avatar: String = ""
    var role: String = "member"     // owner | admin | member
    var nickname: String?

    var displayName: String { (nickname?.isEmpty == false ? nickname! : username) }

    enum CodingKeys: String, CodingKey { case id, username, avatar, role, nickname }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        username = (try? c.decode(String.self, forKey: .username)) ?? ""
        avatar = (try? c.decode(String.self, forKey: .avatar)) ?? ""
        role = (try? c.decode(String.self, forKey: .role)) ?? "member"
        nickname = try? c.decode(String.self, forKey: .nickname)
    }
}

/// GET conversation/{id}/info
struct GroupInfo: Decodable {
    let id: String
    var name: String = ""
    var avatar: String = ""
    var announcement: String = ""
    var ownerId: String = ""
    var myRole: String = "member"
    var members: [GroupMember] = []

    var canManage: Bool { myRole == "owner" || myRole == "admin" }
    func myNickname(_ myId: String) -> String { members.first { $0.id == myId }?.nickname ?? "" }

    enum CodingKeys: String, CodingKey {
        case id, name, avatar, announcement, myRole, members
        case ownerId = "owner_id"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        name = (try? c.decode(String.self, forKey: .name)) ?? ""
        avatar = (try? c.decode(String.self, forKey: .avatar)) ?? ""
        announcement = (try? c.decode(String.self, forKey: .announcement)) ?? ""
        ownerId = (try? c.decode(String.self, forKey: .ownerId)) ?? ""
        myRole = (try? c.decode(String.self, forKey: .myRole)) ?? "member"
        members = (try? c.decode([GroupMember].self, forKey: .members)) ?? []
    }
}
